tools/codegen/gen_backend_stubs.py

```diff
@@ -0,0 +1,114 @@
+                                 ExternalBackendMetadata, assert_never)
+from tools.codegen.selective_build.selector import SelectiveBuilder
+from tools.codegen.utils import Target, concatMap
+import tools.codegen.dest as dest
+
+def parse_backend_yaml(
+        backend_yaml_path: str,
+        grouped_native_functions: Sequence[Union[NativeFunction, NativeFunctionsGroup]]
+) -> Tuple[str, List[Union[ExternalBackendFunction, ExternalBackendFunctionsGroup]]]:
+    with open(backend_yaml_path, 'r') as f:
+        yaml_values = yaml.load(f, Loader=LineLoader)
+    assert isinstance(yaml_values, dict)
+
+    cpp_namespace = yaml_values['cpp_namespace']
+
+    backend = yaml_values['backend']
+    supported = yaml_values['supported']
+    supported_autograd = yaml_values['autograd']
+
+    metadata: Dict[OperatorName, ExternalBackendMetadata] = {}
+    for op in supported:
+        op_name = OperatorName.parse(op)
+        m = ExternalBackendMetadata(op_name, backend, is_autograd=False)
+        metadata[m.operator] = m
+    for op in supported_autograd:
+        op_name = OperatorName.parse(op)
+        m = ExternalBackendMetadata(op_name, backend, is_autograd=True)
+        metadata[m.operator] = m
+
+    native_functions_map: Dict[OperatorName, NativeFunction] = {
+        f.func.name: f
+        for f in concatMap(lambda f: [f] if isinstance(f, NativeFunction) else list(f.functions()), grouped_native_functions)
+    }
+
+    def native_to_external(
+            g: Union[NativeFunction, NativeFunctionsGroup]
+    ) -> Union[ExternalBackendFunction, ExternalBackendFunctionsGroup]:
+        if isinstance(g, NativeFunction):
+            f = g
+            m = metadata.get(f.func.name, None)
+            return ExternalBackendFunction(f, m)
+        elif isinstance(g, NativeFunctionsGroup):
+            return ExternalBackendFunctionsGroup.from_function_group(g, metadata)
+        else:
+            assert_never(g)
+    for op_name in metadata.keys():
+        if op_name not in native_functions_map:
+            raise AssertionError(f"Found an invalid operator name: {op_name}")
+    return cpp_namespace, [native_to_external(g) for g in grouped_native_functions]
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description='Generate backend stub files')
+    parser.add_argument(
+        '-s',
+        '--source_yaml',
+        help='path to source yaml file containing operator external definitions')
+    parser.add_argument(
+        '-o', '--output_dir', help='output directory')
+    options = parser.parse_args()
+
+    # Assumes that this file lives at PYTORCH_ROOT/tools/codegen/gen_backend_stubs.py
+    pytorch_root = pathlib.Path(__file__).parent.parent.parent.absolute()
+    template_dir = os.path.join(pytorch_root, "aten/src/ATen/templates")
+
+    def make_file_manager(install_dir: str) -> FileManager:
+        return FileManager(install_dir=install_dir, template_dir=template_dir, dry_run=False)
```

Comment:
NB: file manager is pointless unless you let people `dry_run` it. The abstraction solely exists to let people dry run the codegen script so that they can find out what files it writes out.

Reply:
I thought FileManager (in non-dry-run mode) also suppressed output if it was unchanged? (Not that supporting dry_run here is a bad idea)

Reply:
ah, got it. It doesn't seem unreasonable to allow an optional `dry_run` flag to the external codegen, similar to what `gen.py` has.